artemis-server/src/test/java/org/apache/activemq/artemis/core/persistence/impl/journal/OperationContextUnitTest.java

```diff
@@ -292,6 +292,97 @@ public void done() {
       }
    }
 
+   @Test
+   public void testIgnoreReplication() throws Exception {
+      ExecutorService executor = Executors.newCachedThreadPool(ActiveMQThreadFactory.defaultThreadFactory(getClass().getName()));
```

Comment:
Could these tests work with a 'same thread executor' (e.g Runnable::run), just to verify that the executor is not called when not expected, and then all the work is done when expected, without any of the waits?